File: Python/me.neonix.versionmanager/versionmanager/version_manager.py
from enum import Enum
from event import Event
from local import Local
from remote import Remote

class Result(Enum):
    OUTDATED = -1,
    CURRENT = 0,
    DEV = 1


class VersionManager:
    __slots__ = ['__local', '__remote', '__status', '__author', '__projectName', '__separator', 'outdated_event', 'current_event', 'dev_event']


    def __init__(self, author: str, projectName: str, version: str, separator='.'):
        self.__author = author
        self.__projectName = projectName
        self.__separator = separator

        self.__local = Local(author, projectName, version, separator)
        self.__remote = None

        self.outdated_event = Event()
        self.current_event = Event()
        self.dev_event = Event()


    def check_status(self):
        if self.__remote is None:
            self.__remote = Remote(self.__author, self.__projectName, self.__separator)
        else:
            self.__remote.update_version()

        if self.__local.verison() == self.__remote.verison():
            self.current_event()
            return Result.CURRENT
        elif self.__local.verison() < self.__remote.verison():
            self.outdated_event()
            return Result.OUTDATED
        else:
            self.dev_event()
            return Result.DEV



vm = VersionManager('Aquatic-Labs', 'Umbra-Mod-Menu', '2.0.5')
vm.dev_event += lambda: print('Dev version')
vm.outdated_event += lambda: print('Outdated version')
vm.current_event += lambda: print('Current version')
vm.check_status()

File: Java/me.neonix.versionmanager/versionmanager/VersionManager.java
import java.util.HashSet;

public class VersionManager {
    private String author;
    private String projectName;
    private String separator;

    private Local local;
    private Remote remote;

    private HashSet<String> observers;

    public VersionManager(String author, String projectName, String version, String separator) {
        this.author = author;
        this.projectName = projectName;
        this.separator = separator;

        this.local = new Local(author, projectName, version, separator);
        this.remote = null;

        this.observers = new HashSet<>();
    }

    private void notifyObservers(HashSet<String> observers) {
        for (String observer : observers) {
            //observer.notify();
        }
    }

    public void registerObserver(String observer) {
        this.observers.add(observer);
    }

    public Status checkStatus() {
        Status status = null;

        if (this.remote == null) {
            this.remote = new Remote(this.author, this.projectName, this.separator);
        } else {
            this.remote.refresh();
        }

        if (this.local.getVersion().compareTo(this.remote.getVersion()) == 0) {
            status = Status.CURRENT;
        } else if (this.local.getVersion().compareTo(this.remote.getVersion()) < 0) {
            status = Status.OUTDATED;
        } else {
            status = Status.DEV;
        }
        this.notifyObservers(this.observers);
        return status;
    }
}